app/javascript/extensible-components/index.ts

```diff
@@ -0,0 +1,67 @@
+import { IExtensionComponent, IMiQApiCallback } from './lib';
+
+/**
+ * Class for easy creation of extensible component.
+ */
+export class ExtensibleComponent {
+  public unsubscribe: Function;
+  constructor(public name: string, public api: IMiQApiCallback, public render: IMiQApiCallback){}
+}
+
+/**
+ * Create new object which will hold extension components on MiQ main object.
+ */
+ManageIQ.extensionComponents = ManageIQ.extensionComponents || {};
```

Comment:
I'd suggest using a simpler name, e.g. `ManageIQ.extensions`.

This extension API can be used by both visual and non-visual components, which can originate from both core and plugins.

Reply:
I'd also suggest to have the MiQ extension API expose a minimal usable set of objects/functions.

Just like with proposed Redux changes [here](https://github.com/vojtechszocs/manageiq-ui-classic/commit/f57df760e4b6d76bb5bb209be428d777f344dc68#diff-2ccaa88a96f0d9b486b557934fb42bfc):

```
ManageIQ.redux = {
  store,
  addReducer,
  applyReducerHash
};
```

Notice that in above snippet, we don't expose impl. details like `rootReducer` etc.